addons/hr_work_entry/models/hr_work_entry.py

```diff
@@ -79,14 +82,15 @@ def _check_if_error(self):
             return False
         undefined_type = self.filtered(lambda b: not b.work_entry_type_id)
         undefined_type.write({'state': 'conflict'})
-        conflict = self._mark_conflicting_work_entries(min(self.mapped('date_start')), max(self.mapped('date_stop')))
+        conflict = self._mark_conflicting_work_entries(min(self.mapped('date_start')), max(self.mapped('date_stop')), self.ids)
         return undefined_type or conflict
 
     @api.model
-    def _mark_conflicting_work_entries(self, start, stop):
+    def _mark_conflicting_work_entries(self, start, stop, ids=()):
```

Comment:
is this allowed?

Reply:
Probably not :(
On the flip side this method is called only in this file. There is also [this](https://github.com/odoo/enterprise/blob/245293a6e37fbe881154bc3d84a6ba44445621b7/test_l10n_be_hr_payroll_account/__init__.py#L65) comment that points to a similar issue.

Reply:
Updated to use the ids from self as per @KangOl suggestion ;) [not tested yet]